test/if_lub_gepinst_test/index.c

```diff
@@ -0,0 +1,35 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "../../Annotations/Annotations.h"
+
+struct M {
+    char* x MustCall("free, free1");
+    char* y MustCall("free");
+};
+
+void free1(void* p Calls("free1")) {
+    free(p);
+}
+
+int main() {
+    struct M s;
+
+    s.x = (char*)malloc(15);
+
+    s.y = s.x; // note: when we alias s.y and s.x, the mc of the set in total will be {"free", "free1"}
+    // how does this happen? we *first* do the alias reasoning and find out s.x <-> s.y,
```

Comment:
Is the alias reasoning correctly combining the must-call sets? In the Java implementation, we wouldn't even permit this assignment (because x's `MustCall(free, free1)` is not a subtype of y's `MustCall(free1)` (assignments in the other direction would be permitted)). With that context, I think the behavior here is correct: if we assign `x` to `y`, the safe thing to do is to take the worst case scenario (we have to fulfill `x`'s must-call obligation on `y`)

Reply:
We may also want to discuss this in detail.

Reply:
The assignment is improperly allowed here since annotation verification is not yet implemented; I did not catch the error myself. I'll update the test to be logically consistent